### ISSUE TITLE
Provide better resilience regarding the draft storage

### ISSUE THREAD

**Issue by guimanjosh** [eed-web-application/elog-plus#210]
Consider the following user story:

1. User makes a draft to logbook A that they have write access to
2. An admin revokes write access to logbook A from the user
3. User goes back to the new entry page with the saved draft

The draft is stale since the user no longer has write access to the logbook. Right now, the page just crashes.

There will be many issues like this since the draft storage is not reactive to the backend, so we must have resilience to stale data, and figure out what to do in these cases like above.

Probably won't fix, because the backend should handle drafts at some point anyways.

-Boogie

**Comment by bisegni** [comment]
@eed-web-application/eed-deployer @guimanjosh want we start to introduce the draft `management/per user` on backend?

**Comment by guimanjosh** [comment]
Yes, should introduce `draft management/per user` on the backend. I think it will be more secure and consistent if draft `management/per user` is handled on the backend.

**Comment by bisegni** [comment]
explain what should be the proposed workflow in this situation

**Comment by guimanjosh** [comment]
1. **User Authentication and Authorization:**
   - Users authenticate themselves to access the application.
   - Admins have additional privileges to manage user permissions.
   - Backend verifies user credentials and roles upon each request.

2. **Draft Creation:**
   - When a user creates a new draft, send a request to the backend to create a draft object associated with the user.
   - Admins can set initial permissions for the draft (e.g., read-only, read-write).

3. **Draft Editing:**
   - As the user edits the draft, send periodic updates to the backend.
   - Backend validates permissions on each edit to ensure the user has the necessary access rights.

4. **Permission Changes by Admin:**
   - If an admin changes a user's permissions while they are editing a draft, consider the following:
     - If permissions are upgraded (e.g., read-only to read-write), allow continued editing.
     - If permissions are downgraded (e.g., read-write to read-only), notify the user and prevent further edits.

5. **Draft Retrieval:**
   - When a user requests their drafts, retrieve them from the backend, considering their permissions.
   - Admins have access to all drafts and can manage permissions accordingly.

6. **Draft Updating:**
   - Users can update their drafts within their permitted access levels.
   - Backend validates each update against the user's current permissions.

7. **Draft Deletion:**
   - Users can delete their drafts if they have the necessary permissions.
   - Admins can delete drafts of any user based on their administrative privileges.

8. **Error Handling:**
   - Implement error handling for scenarios like network errors, permission changes, or validation failures.
   - Provide clear error messages to users and admins to guide them on how to proceed.

9. **Version Control and Audit Trails:**
   - Maintain a version history for each draft, tracking changes made by users and admins.
   - Include metadata like timestamps and user/admin IDs for accountability?

10. **Offline Support:**
    - Provide limited offline support based on permissions and caching mechanisms.
    - Notify users if they try to edit drafts while offline and their permissions have changed.
